### PR TITLE
Add GIDToken Class

### PR DESCRIPTION
**Pull request by Alex-4-Git**
This class contains a token string property and an optional expiration date property.

### CODE REVIEW

File: GoogleSignIn/Sources/GIDToken.m [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Google LLC
```
Comment (petea) [inline GoogleSignIn/Sources/GIDToken.m:2]: "2022" here and throughout.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDToken.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// This class represents the basic information fo a token.
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDToken.h:21]: "fo" -> "of"

File: GoogleSignIn/Sources/GIDToken.m [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+#import "GoogleSignIn/Sources/GIDToken_Private.h"
+
+// Key constants used for encode and decode.
+static NSString *const kTokenStringKey = @"tokebString";
+static NSString *const kExpirationDateKey = @"expirationDate";
+
+@implementation GIDToken
+
+- (instancetype)initWithTokenString: (NSString *)tokenString
```
Comment (petea) [inline GoogleSignIn/Sources/GIDToken.m:27]: No space between `:` and `(`, here and throughout.

File: GoogleSignIn/Tests/Unit/GIDTokenTest.m [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+#import <XCTest/XCTest.h>
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDToken.h"
+
+#import "GoogleSignIn/Sources/GIDToken_Private.h"
+
+@interface GIDTokenTest : XCTestCase
+@end
+
+@implementation GIDTokenTest
```
Comment (petea) [inline GoogleSignIn/Tests/Unit/GIDTokenTest.m:23]: Add an initializer test?

**Review by petea — CHANGES_REQUESTED** [review]

File: GoogleSignIn/Sources/GIDToken.m (hunk 1 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/GIDToken.m:2]: Done.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDToken.h (hunk 2 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDToken.h:21]: Done.

File: GoogleSignIn/Sources/GIDToken.m (hunk 3 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/GIDToken.m:27]: Done.

File: GoogleSignIn/Tests/Unit/GIDTokenTest.m (hunk 4 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Tests/Unit/GIDTokenTest.m:23]: Done.

**Review by mdmathias — APPROVED** [review]

**Pull request merged**